### PR TITLE
feat: improve textual output

### PR DESCRIPTION
**Pull request by fnaranjo-vmw**
I strongly recommend reviewing this PR commit by commit, instead of looking at "Files changed".

### Checklist:

* [ ] Have you added Release Notes in the docs respositories?
* [x] Have you followed the [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/#summary)?

### CODE REVIEW

File: internal/upgrader/upgrader_test.go [hunk 1]
```diff
@@ -120,12 +132,12 @@ var _ = Describe("Upgrade", func() {
 
 	When("performing a dry run", func() {
 		It("should print out service GUIDs and not attempt to upgrade", func() {
-			var lines []string
-			fakeLog.PrintfStub = func(format string, a ...any) {
-				lines = append(lines, fmt.Sprintf(format, a...))
-			}
+			logr := logger.New(time.Minute)
```
Comment (pivotal-marcela-campo) [inline internal/upgrader/upgrader_test.go:135]: note: Can we next time do this kind of improvements in a separate PR? Just so it is easier to review.

File: internal/logger/logger_test.go [hunk 2]
```diff
@@ -16,6 +17,37 @@ import (
...
 
+func testPlan0() ccapi.Plan {
+	return ccapi.Plan{
+		GUID:                   "fake-plan-guid",
+		Name:                   "fake-plan-name",
+		MaintenanceInfoVersion: "fake-plan-version",
+		ServiceOfferingGUID:    "fake-service-offering-guid",
+	}
+}
+
```
Comment (pivotal-marcela-campo) [inline internal/logger/logger_test.go:28]: suggestion(blocking): testInstance2 and testInstance5 are not descriptive. I thought there was an array of some kind. And also one calls the other.

**Review by pivotal-marcela-campo — CHANGES_REQUESTED** [review]

File: internal/upgrader/upgrader.go [hunk 3]
```diff
@@ -70,12 +71,14 @@ func performUpgrade(api CFClient, upgradableInstances []ccapi.ServiceInstance, p
 		for instance := range upgradeQueue {
 			start := time.Now()
 			log.UpgradeStarting(instance.ServiceInstance, instance.Plan)
-			err := api.UpgradeServiceInstance(instance.ServiceInstance.GUID, instance.Plan.MaintenanceInfoVersion)
-			switch err {
-			case nil:
-				log.UpgradeSucceeded(instance.ServiceInstance, instance.Plan, time.Since(start))
-			default:
-				log.UpgradeFailed(instance.ServiceInstance, instance.Plan, time.Since(start), err)
+			if !dryRun {
```
[the remark this answers is not in the document]
Comment (fnaranjo-vmw) [reply internal/upgrader/upgrader.go]: Force-pushed a new PR history. (The 3 commits are different)
This issue is addressed in the new PR.

File: internal/upgrader/upgrader_test.go (hunk 1 above)
Comment (fnaranjo-vmw) [reply internal/upgrader/upgrader_test.go]: Force-pushed a new PR history. (The 3 commits are all different)

As we have discussed in via Zoom, this isn't a aesthetic improvement.
`fakeLog.PrintfStub` prevented some new code from being tested correctly.

This issue is addressed in the new commit history.

File: internal/logger/logger_test.go (hunk 2 above)
Comment (fnaranjo-vmw) [reply internal/logger/logger_test.go]: This issue has been addressed in the new commit history.

File: internal/logger/logger_test.go [hunk 4]
```diff
@@ -42,53 +43,57 @@ var _ = Describe("Logger", func() {
...
 		Expect(result).To(MatchRegexp(timestampRegexp + `: upgrade of instance: "my-instance" guid: "fake-guid" failed after 1m0s: boom\n`))
 	})
 
 	It("can log the final totals", func() {
 		l.InitialTotals(10, 5)
-		l.UpgradeFailed("my-first-instance", "fake-guid-1", time.Minute, fmt.Errorf("boom"))
-		l.UpgradeFailed("my-second-instance", "fake-guid-2", time.Minute, fmt.Errorf("bang"))
-		l.UpgradeSucceeded("my-third-instance", "fake-guid-3", time.Minute)
-		l.SkippingInstance("skipped", "skipped-guid", true, "create", "failed")
+		l.UpgradeFailed(basicInstance("my-first-instance", "fake-guid-1"), time.Minute, fmt.Errorf("boom"))
```
Comment (pivotal-marcela-campo) [inline internal/logger/logger_test.go:74]: nitpick: we are logging upgrade failed for an instance that doesn't have an upgrade available. If it is easy to pass the upgrade flag in true for these scenarios I would very much prefer that.

**Review by pivotal-marcela-campo — APPROVED** [review]

File: internal/logger/logger_test.go [hunk 5]
```diff
@@ -42,53 +43,57 @@ var _ = Describe("Logger", func() {
...
+		l.UpgradeFailed(basicInstance("my-second-instance", "fake-guid-2"), time.Minute, fmt.Errorf("bang"))
```
Comment (fnaranjo-vmw) [inline internal/logger/logger_test.go:75]:
```suggestion
		l.UpgradeFailed(fullInstance("my-first-instance", "fake-guid-1", true, "fake-op-type1", "fake-op-state1"), time.Minute, fmt.Errorf("boom"))
		l.UpgradeFailed(fullInstance("my-second-instance", "fake-guid-2", true, "fake-op-type2", "fake-op-state2"), time.Minute, fmt.Errorf("bang"))
```

File: internal/logger/logger_test.go (hunk 4 above)
Comment (fnaranjo-vmw) [reply internal/logger/logger_test.go]: Fixed by latest commit: https://github.com/cloudfoundry/upgrade-all-services-cli-plugin/pull/89/commits/daca87dd469c4ad258ae5da7f5d866bf2850c5f9
